### DISCUSSION TITLE
how to calculate Precision/Recall/Accuracy in v1.1.10?

### DISCUSSION THREAD

**Discussion by stevensagaert** [open-edge-platform/anomalib#2213] [Q&A] [answered]
### Discussed in https://github.com/openvinotoolkit/anomalib/discussions/781

When I do 
engine = Engine(max_epochs=epochs, task=TaskType.CLASSIFICATION,image_metrics=["Precision","Recall","Accuracy"])
I get the following output:
Incorrect constructor arguments for Precision metric from TorchMetrics package.
Incorrect constructor arguments for Recall metric from TorchMetrics package.
Incorrect constructor arguments for Accuracy metric from TorchMetrics package

**Comment by samet-akcay** [comment]
@stevensagaert, what torchmetrics version are you using? Also, can you provide the full error message you are seeing

**Comment by stevensagaert** [comment]
1.4.0.post0

The error message is was I gave. The code continues.
________________________________

**Comment by ashwinvaidya17** [comment, marked as answer]
@stevensagaert newer version of Torchmetrics require you to also pass the `task` parameter. One solution would be to use dict to pass the argument

```python
from anomalib.data import MVTec
from anomalib.engine import Engine
from anomalib.models import Padim

if __name__ == "__main__":
    model = Padim()
    data = MVTec()
    engine = Engine(
        image_metrics={
            "Precision": {
                "class_path": "torchmetrics.Precision",
                "init_args": {"task": "binary"},
            }
        },
    )
    engine.train(model, datamodule=data)
```
<img width="491" alt="afbeelding" src="https://github.com/user-attachments/assets/aacc4a02-80fc-48d1-9896-2960ee0e54a1">

**Reply by mykappa** [reply]
How can I pass the task argument to the metric when using anomalib's CLI and a config.yml?

**Comment by stevensagaert** [comment]
Got it. Thanks.
________________________________

**Reply by ashwinvaidya17** [reply]
@stevensagaert once this PR https://github.com/openvinotoolkit/anomalib/pull/2212 is merged you can directly pass the metric objects to the Engine

**Comment by mohandeeprdori** [comment]
@ashwinvaidya17 Hi,
How do i get to print these metrics when I'm doing engine.predict(). Below is my code. When i ran this code. It did not print anything. 
```

from torchmetrics.classification import Accuracy, Precision, Recall

from anomalib.data import MVTec
from anomalib.engine import Engine
from anomalib.models import Fastflow

if __name__ == "__main__":
    model = Fastflow()
    datamodule = MVTec()
    ckpt_path = './anomalib/results/Fastflow/MVTec/bottle/v1/weights/lightning/model.ckpt'
    # engine = Engine(image_metrics=[Accuracy(task="binary"), Precision(task="binary"), Recall(task="binary")])
    engine = Engine(
        image_metrics={
            "Precision": {
                "class_path": "torchmetrics.Precision",
                "init_args": {"task": "binary"},
            },
            "Recall": {
                "class_path": "torchmetrics.Recall",
                "init_args": {"task": "binary"},
            },
            "Accuracy": {
                "class_path": "torchmetrics.Accuracy",
                "init_args": {"task": "binary"},
            }
        },
    )
    engine.predict(datamodule=datamodule, model=model, ckpt_path=ckpt_path,)
    # print("Predictions:",predictions)

```


print(predictions) gave me tensors and masks which i dont require at this point.

**Reply by ashwinvaidya17** [reply]
Predict is used just to do inference. So, that's why you get tensors and mask. If you want metrics, you need to call `Engine.test`. Keep in mind, the `test_dataloader` requires ground truth masks as well.

**Comment by mohandeeprdori** [comment]
Thanks @ashwinvaidya17 it was helpful. I was able to get the metrics as needed. Is it possible to get metrics for each image or each class?
